tools/clang/lib/SPIRV/SpirvBuilder.cpp

```diff
@@ -980,12 +974,70 @@ SpirvBuilder::createBitFieldInsert(QualType resultType, SpirvInstruction *base,
   return inst;
 }
 
-SpirvBitFieldExtract *SpirvBuilder::createBitFieldExtract(
-    QualType resultType, SpirvInstruction *base, SpirvInstruction *offset,
-    SpirvInstruction *count, bool isSigned, SourceLocation loc) {
+SpirvInstruction *SpirvBuilder::createEmulatedBitFieldExtract(
```

Comment:
Do we need to emulate `OpBitFieldUExtract` differently when `isSigned = false`?

Reply:
I don't think so:
If the type was unsigned, we wouldn't have to do sign extension, to masking could be enough. But this requires also 2 instructions (shift + masking).
Since using 2 shifts should yield the same result, I think both can use this method.

(Changed the instruction from ShiftRightLogical to ShiftRightArithmetic, as the second does sign-extension when required vs the first)

Reply:
Sorry, I wasn't very clear. I meant that the `createBitFieldExtract` function has a distinct `isSigned` parameter, and for non-emulated versions, we determine whether or not to do sign extension based on its value, but for the emulated version, it's now based on the signedness of `base`.

If we know those are equivalent for all the existing usages, we should remove the `isSigned` param because it's now a misleading interface. If we're not certain, then we should pass `isSigned` onto `createEmulatedBitFieldExtract` and use it there too.

Reply:
Right, got it.
I checked, and it is invalid to have the resultType != baseType on OpBitFieldExtract instructions. This means we can use the AST result type to get the `isSigned`, and there is no risk of mismatch.
Cleanup those.